tests/compiletest.rs

```diff
@@ -15,7 +15,7 @@ fn run_mode(mode: &'static str) {
             .expect("need to specify RUST_SYSROOT env var or use rustup or multirust")
             .to_owned(),
     };
-    let sysroot_flag = format!("--sysroot {}", sysroot);
+    let sysroot_flag = format!("--sysroot {} -Dwarnings", sysroot);
```

Comment:
so our compile-fail tests don't contain unnecessary warnings anymore... like functions that are never called (those had `#[miri_run]`, but weren't called from `main`)